web-client/src/providers/realUserMonitoring.ts

```
@@ -0,0 +1,30 @@
+import { AwsRum, AwsRumConfig } from 'aws-rum-web';
+
+export const initializeRealUserMonitoring = () => {
+  if (process.env.ENV === 'local') return;
+  let awsRum: AwsRum;
+  try {
+    const config: AwsRumConfig = {
+      allowCookies: true,
+      enableXRay: false,
+      endpoint: 'https://dataplane.rum.us-east-1.amazonaws.com',
+      identityPoolId: process.env.RUM_IDENTITY_POOL_ID,
+      sessionSampleRate: process.env.ENV === 'prod' ? 0.1 : 1, // sample 10% of sessions in production until we know how expensive this will be, 100% in lower environments
+      telemetries: ['performance'],
+    };
+
+    const APPLICATION_ID: string = process.env.RUM_APP_MONITOR_ID!;
+    const APPLICATION_VERSION: string = '0.0.1';
+
+    awsRum = new AwsRum(
+      APPLICATION_ID,
+      APPLICATION_VERSION,
+      'us-east-1',
+      config,
+    );
+
+    awsRum.enable();
+  } catch (error) {
+    // Ignore errors thrown during CloudWatch RUM web client initialization
```

We ignore errors on RUM Client init so that we don't disruptions for users if the client fails to initialize for some reason.

---

Is it worth logging this failure?

---

I went back-and-forth on this. It might have some use, at least for us in lower environments, but in production it's going to be logging errors to the user's browser console, which we're not currently collecting anywhere (I don't think).

My concern with turning it on in production is that I don't know what it might log. I would hope that it wouldn't end up logging anything sensitive, but I can't with 100% confidence say it wouldn't.

I'm on the fence, and could pretty easily be convinced to go ahead and add a log here.

---

I went ahead and added the log. After some thought, the potential for it to be useful for troubleshooting does seem worth it enough to me.

---

Oh, I was thinking just calling the [logError endpoint](https://github.com/ustaxcourt/ef-cms/blob/57d0fce3ff870699f7c0a6a64dbcced362973f8d/web-api/src/app.ts#L1090) with a generic message like `RUM failed to initialize` so we could monitor the monitoring lol. Whatever you decide is fine, I don't have strong feelings.